src/mapgen.cpp

```diff
@@ -6725,10 +6726,22 @@ void map::rotate( int turns )
             old_y += SEEY;
         }
 
-        const auto new_pos = point{ old_x, old_y } .rotate( turns, { SEEX * 2, SEEY * 2 } );
-
-        np.spawn_at_precise( { abs_sub.xy() }, { new_pos, abs_sub.z } );
-        overmap_buffer.insert_npc( npc_ptr );
+        const point new_pos = point{ old_x, old_y } .rotate( turns, { SEEX * 2, SEEY * 2 } );
+        if( setpos_safe ) {
+            // setpos can't be used during mapgen, but spawn_at_precise clips position
+            // to be between 0-11,0-11 and teleports NPCs when used inside of update_mapgen
+            // calls
+            const tripoint new_global_sq = sq - local_sq + new_pos;
+            const tripoint &new_local_pos = g->m.getlocal( new_global_sq );
+            np.setpos( tripoint( new_local_pos ) );
```

Comment:
Why does this make an explicit copy of `new_local_pos`?
```suggestion
            np.setpos( new_local_pos );
```

Reply:
I have no idea, thanks for the review.